vignettes/deepdive.Rmd

```diff
@@ -0,0 +1,404 @@
+
+## SDTM and ADaM Data Packages
+
+__SDTM:__ The primary pieces of the SDTM package are the SDTM annotated case report forms (acrf.pdf), the data definitions document (define.xml), the Study Data Reviewer's Guide (sdrg.pdf) and the datasets in xpt Version 5 format.  The Version 5 xpt file is the **required** submission format for all datasets going to the Health Authorities.  
+
+__ADaM:__ The key components of the ADaM package are very similar to SDTM package with a few additions: define.xml, Analysis Study Data Reviewer's Guide (adrg.pdf), Analysis Results Metadata (analysis-results-metadata.pdf) and datasets as Version 5 xpt format.
+
+As both Data Packages need compliant `xpt` files, we feel that `{xportr}` can play a pivotal role here. The core functions in `{xportr}` can be used to apply information from the _metadata object_ to the datasets giving users feedback on the quality of the metadata and data. `xportr_write()` can then be used to write out the final dataset as an `xpt` file, which can be submitted to a Health Authority.
+
+## What is `{xportr}` validating in these Data Packages?
+
+The `xpt` Version 5 files form the backbone of any successful Submission and are govern by quite a lot of rules and suggested guidelines. As you are preparing your data packages for submission the suite of core `{xportr}` functions, plus `xportr_write()`, helps to check that your datasets are submission compliant. The package checks many of the latest rules laid out in the [Study Data Technical Conformance Guide](https://www.fda.gov/regulatory-information/search-fda-guidance-documents/study-data-technical-conformance-guide-technical-specifications-document), but please note that it is not yet an exhaustive list of checks. We envision that users are also submitting their `xpts` and metadata to additional validation software.
+
+Each of the core `{xportr}` functions for applying labels, types, formats, order and lengths provides feedback to users on submission compliance. However, a final check is implemented when `xportr_write()` is called to create the `xpt`. `xportr_write()` calls [`xpt_validate()`](https://github.com/atorus-research/xportr/blob/231e959b84aa0f1e71113c85332de33a827e650a/R/utils-xportr.R#L174), which is a behind the scenes/non-exported function that does a final check for compliance.  At the time of `{xportr} v0.3.0` we are checking the following when a user writes out an `xpt` file.:
+
+<img src="xpt_validate.png" alt="validate" style="width:800px;"/>
+
+
+# {xportr} in action
+
+In this section, we are going to explore the 5 core `{xportr}` functions using: 
+
+* An ADSL ADaM dataset from the Pilot 3 Submission to the FDA
+* The ADSL ADaM Specification File from the Pilot 3 Submission to the FDA
```

Comment:
```suggestion
* `xportr::adsl` - An ADSL ADaM dataset from the Pilot 3 Submission to the FDA
* `xportr::var_spec` - The ADSL ADaM Specification File from the Pilot 3 Submission to the FDA
```
It would be nice to have a link to the data source where it is pointed out. WDYT?

Reply:
Yes this is a great idea!!